src/backend/catalog/gp_matview_aux.c

```diff
@@ -290,6 +296,10 @@ RemoveMatviewTablesEntries(Oid mvoid)
 		if (!HeapTupleIsValid(tuple))
 			continue;
 
+		/* update relation's pg_class entry */
+		relid = ((Form_gp_matview_tables) GETSTRUCT(tuple))->relid;
```

Comment:
how to handle partitioned table ?

Reply:
Same approach, no difference. What's your specific concern?